server/src/main/java/org/elasticsearch/index/engine/VersionConflictEngineException.java

```diff
@@ -30,6 +31,16 @@ public VersionConflictEngineException(ShardId shardId, Engine.Operation op, long
         this(shardId, op.type(), op.id(), op.versionType().explainConflictForWrites(currentVersion, op.version(), deleted));
     }
 
+    public VersionConflictEngineException(ShardId shardId, String type, String id,
```

Comment:
I think we should rename it to `EngineConflictException` the prefix `Elasticseach` is obsolete